### PR TITLE
Added exception cause to logs and exception email

### PR DESCRIPTION
**Pull request by Dantemss**
- Exception causes are now printed in logs and emails, if present
- First line of backtrace added to main exception and cause
- Specs

### CODE REVIEW

File: config/initializers/controllers.rb [hunk 1]
```diff
@@ -21,6 +21,17 @@ def set_app_date_header
...
 
+  def get_exception_cause(exception)
+    cause = exception.cause
+    return if cause.nil?
+
+    {
+      class: cause.class.name,
+      message: cause.message,
+      cause: get_exception_cause(cause)
+    }
```
Comment (jpslav) [inline config/initializers/controllers.rb]: Can you add at least the first line of the backtrace (where the causing exception was raised?)

File: config/initializers/controllers.rb [hunk 2]
```diff
@@ -67,7 +78,9 @@ def rescue_from_exception(exception)
         env: request.env,
         data: {
           error_id: @error_id,
+          class: exception.class.name,
           message: exception.message,
+          cause: get_exception_cause(exception),
```
Comment (jpslav) [inline config/initializers/controllers.rb]: Can you add the cause to the log message too?

File: spec/controllers/application_controller_spec.rb [hunk 3]
```diff
@@ -122,4 +122,126 @@
       expect(header_time).to be_within(1.second).of(t)
     end
   end
+
+  context 'rescue_from_exception' do
```
Comment (jpslav) [inline spec/controllers/application_controller_spec.rb]: Thanks!

File: config/initializers/controllers.rb (hunk 1 above)
Comment (Dantemss) [inline config/initializers/controllers.rb]: Want this on the main one too?

File: config/initializers/controllers.rb (hunk 2 above)
Comment (Dantemss) [inline config/initializers/controllers.rb]: What part? Or the whole thing?

File: config/initializers/controllers.rb (hunk 1 above)
Comment (jpslav) [inline config/initializers/controllers.rb]: The main one has the full backtrace via ExceptionNotifier, I believe.

File: config/initializers/controllers.rb (hunk 2 above)
Comment (jpslav) [inline config/initializers/controllers.rb]: The whole thing, class, message, first line of backtrace, next level down cause.

File: config/initializers/controllers.rb (hunk 1 above)
Comment (Dantemss) [inline config/initializers/controllers.rb]: Hm I like consistency though so I think I'll add it there too :)

**Pull request merged**